### PR TITLE
Document new process instance migration restrictions

### LINKED ISSUE #4186: Document new process instance migration restrictions for boundary events
With 8.6, it's possible to map boundary events (and thus migrate their respective event subscriptions). While this adds more possibilities, we added several new restrictions to these new options. They should of course be documented in the [API](https://docs.camunda.io/docs/apis-tools/zeebe-api/gateway-service/#errors-9) and the [concept limitations](https://docs.camunda.io/docs/components/concepts/process-instance-migration/#limitations): 

- https://github.com/camunda/camunda/issues/19956
- https://github.com/camunda/camunda/issues/19960

For the API, it's easiest to just align it directly with the [`gateway.proto`](https://github.com/camunda/camunda/blob/main/zeebe/gateway-protocol/src/main/proto/gateway.proto) file.

For the limitations, we should describe these cases in the first section (limits that won't be lifted in the future).

### PR DESCRIPTION
**Pull request by korthout**
## Description

In 8.6, process instance migration supports the mapping of boundary events. While this adds a whole new level of migration capabilities, we've also had to add some additional restrictions to avoid the migration from adjusting the process instance to a state where it cannot continue execution correctly.

See:
- https://github.com/camunda/camunda/issues/19956
- https://github.com/camunda/camunda/issues/19960

This Pull Request adjusts both the gRPC API documentation as well as the Concept documentation to allow mapping boundary events.

No example has been added to the Concept documentation yet that showcases the behavior and use case of mapping a boundary event during process instance migration. This will be done later, probably together with the addition of the timer boundary event support.

closes #4186 

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/concepts/process-instance-migration.md [hunk 1]
```diff
@@ -202,17 +202,13 @@ After migrating active element `A`, the process instance is no longer subscribed
...
 :::
 
 :::tip
-Currently, you cannot migrate an active element with a message boundary event attached to an element that also has a message boundary event attached with the same message name.
-While we're working on resolving this, a workaround is available by splitting the migration into two steps:
-
-First migrate the process instance to a process model where the element is without a message boundary event.
-Once the process instance is successfully migrated, you can migrate the process instance to a process model where the message boundary event is attached to the element again.
+Currently, you cannot migrate an active element with a message boundary event attached to an element that also has a message boundary event attached if both the boundary events rely on the same message name and no mapping is provided between these boundary events.
+While we're working on resolving this, you can migrate this case by providing a mapping between the boundary events.
```
Comment (christinaausley) [inline docs/components/concepts/process-instance-migration.md:211]: Do you mean "While we're working on resolving this, you can **mitigate** this case by providing a mapping between the boundary events."?

Comment (korthout) [reply docs/components/concepts/process-instance-migration.md:211]: @christinaausley migrate here referred to the topic `process instance migration`. The intention is to describe a workaround so users can still migrate their process instance even with this limitation. Alternative wording could be:

```
While we're working on resolving this, you can provide a mapping between the boundary events to workaround this limitation.
```

File: docs/components/concepts/process-instance-migration.md [hunk 2]
```diff
@@ -227,6 +223,8 @@ In the following cases, the process instance can't apply the migration plan and
 - The migration plan can only map each `sourceElementId` once.
 - A mapping instruction's `sourceElementId` must refer to an element existing in the process instance's process definition.
 - A mapping instruction's `targetElementId` must refer to an element existing in the target process definition.
+- A mapping instruction cannot detach a boundary event from an active element.
```
Comment (berkaycanbc) [inline docs/components/concepts/process-instance-migration.md:226]: 🔧 We can be more explicit here: "A mapping instruction provided between boundary events cannot detach a boundary event from an active element."

File: docs/components/concepts/process-instance-migration.md [hunk 3]
```diff
@@ -227,6 +223,8 @@ In the following cases, the process instance can't apply the migration plan and
...
+- Each boundary event can only be the target of a mapping instruction once.
```
Comment (berkaycanbc) [inline docs/components/concepts/process-instance-migration.md:227]: ❌ This sentence is too implicit to understand. An alternative would be: "Multiple boundary events in the source cannot be mapped to the same boundary event in the target."

File: docs/components/concepts/process-instance-migration.md [hunk 4]
```diff
@@ -202,17 +202,13 @@ After migrating active element `A`, the process instance is no longer subscribed
...
 Other types of boundary events will be supported in future versions.
 :::
 
 :::tip
-Currently, you cannot migrate an active element with a message boundary event attached to an element that also has a message boundary event attached with the same message name.
-While we're working on resolving this, a workaround is available by splitting the migration into two steps:
-
-First migrate the process instance to a process model where the element is without a message boundary event.
-Once the process instance is successfully migrated, you can migrate the process instance to a process model where the message boundary event is attached to the element again.
+Currently, you cannot migrate an active element with a message boundary event attached to an element that also has a message boundary event attached if both the boundary events rely on the same message name and no mapping is provided between these boundary events.
```
Comment (berkaycanbc) [inline docs/components/concepts/process-instance-migration.md:210]: ❌ This sentence is hard to follow to the end. I think, we can simply add "without providing mapping instructions between boundary events" at the end of the old version. Then it becomes: `Currently, you cannot migrate an active element with a message boundary event attached to an element that also has a message boundary event attached with the same message name without providing mapping instructions between boundary events.`. I am also open for other simpler explanations.

**Review by berkaycanbc — CHANGES_REQUESTED** [review]
@korthout this is a very valuable update in our migration docs, thanks! 🚀 

I requested some changes to better clarify current limitations, please take a look. 👀

**Comment by korthout** [comment]
Thanks all for the reviews. I've come the conclusion that the restrictions need to be generalized anyways. They don't just apply to boundary events. I've marked the PR on hold for now.

**Comment by christinaausley** [comment]
@korthout I will be out for two weeks after tomorrow, so feel free to loop in @camunda/tech-writers for additional review when you're ready 👍

**Comment by mesellings** [comment]
@korthout Did you need this reviewed and merged for the alpha release next week, or is this on hold beyond that?

**Comment by korthout** [comment]
Hi @mesellings 👋 

This is on hold as we were discovering that the rule is wider than just boundary events. Rather they concern all catch events that active elements are subscribed to. While the functionality is now available in alpha5, I don't see a use in getting this merged as is. So, let's hold off on merging until we've adjusted the documentation accordingly.

cc @berkaycanbc

**Comment by mesellings** [comment]
@korthout Thanks for the clarification - we'll hold off for alpha 5, and will perform a review once the docs are adjusted 👍

**Comment by christinaausley** [comment]
@korthout would you like this to go out with release next week, or is this still on hold?

**Comment by korthout** [comment]
@christinaausley Thanks for reminding me. This is superseded by:
- #4348
